pycbc/coordinates/space.py

```diff
@@ -0,0 +1,893 @@
+# Copyright (C) 2023  Shichao Wu, Alex Nitz
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This module provides coordinate transformations related to space-borne
+detectors, such as coordinate transformations between space-borne detectors
+and ground-based detectors.
+"""
+
+import numpy as np
+from scipy.spatial.transform import Rotation
+from scipy.optimize import fsolve
```

Comment:
As mentioned, we should have  unittests for these functions that verify where possibly that they give answers we've validated to be correct and that they are revsesible. Make a new file in the test directory modeled after the test_conversions.py one. It will be automatically run by the test suite. You should first verify it works on your local machine though.

Reply:
@ahnitz I add the comment on https://github.com/gwastro/pycbc/pull/4289/files#diff-7a1da599956f99d94f2c70707533b585de4d61399182d8b58954fe73ed1e15dcR27